broadinstitute/single_cell_portal_core
# Fix perfTime analytics for Study Overview (SCP-3042)

This fixes and enriches low-level client-side analytics for plot performance in Study Overview:

* `plot:scatter`
* `plot:violin`
* `plot:box`
* `plot:dotplot`
* `plot:heatmap`

These correspond to low-level analytics diagrammed in #716.  Analytics for "composite" time-to-interactive logging will be addressed in SCP-2907.  These changes provide granular baseline analytics to assess scalability improvements.  The other issue provide higher-level, summary baseline data for the same.

To test:
* Load Study Overview
* Verify that a `plot:scatter` event is logged to Bard / Mixpanel
* Verify that the logged event includes `perfTime`, `perfTime:backend`, and `perfTime:frontend` properties, and that the latter two have values smaller than the first.

Here's an example of what's logged:
<img width="1680" alt="Screen Shot 2021-03-18 at 2 00 02 PM" src="https://user-images.githubusercontent.com/1334561/111674313-438bc180-87f2-11eb-937a-8399971c0302.png">

This satisfies SCP-3042.
